src/NHibernate/Connection/ResilientDriverConnectionProvider.cs

```diff
@@ -0,0 +1,97 @@
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Data.Common;
+using System.Threading;
+using NHibernate;
+using NHibernate.Connection;
+
+namespace NHibernate.Connection
+{
+	public class ResilientDriverConnectionProvider : DriverConnectionProvider
+	{
+		public const String ConnectionDelayBetweenTries = "connection.delay_between_tries";
+		public const String ConnectionMaxTries = "connection.max_tries";
+
+		private static readonly IInternalLogger log = LoggerProvider.LoggerFor(typeof(ResilientDriverConnectionProvider));
+
+		public ResilientDriverConnectionProvider()
+		{
+			this.MaxTries = 3;
+			this.DelayBetweenTries = TimeSpan.FromSeconds(5);
+		}
+
+		public Int32 MaxTries { get; set; }
+
+		public TimeSpan DelayBetweenTries { get; set; }
+
+		public override void Configure(IDictionary<String, String> settings)
+		{
+			String maxTries;
+			String delayBetweenTries;
+
+			if (settings.TryGetValue(ConnectionMaxTries, out maxTries) == true)
+			{
+				this.MaxTries = Int32.Parse(maxTries);
+			}
+
+			if (settings.TryGetValue(ConnectionDelayBetweenTries, out delayBetweenTries) == true)
+			{
+				TimeSpan timespan;
+				Int32 delay;
+
+				if (TimeSpan.TryParse(delayBetweenTries, out timespan) == true)
+				{
+					this.DelayBetweenTries = timespan;
+				}
+				else if (Int32.TryParse(delayBetweenTries, out delay) == true)
+				{
+					this.DelayBetweenTries = TimeSpan.FromSeconds(delay);
+				}
+				else
+				{
+					throw (new Exception("Invalid delay"));
+				}
+			}
+
+			base.Configure(settings);
+		}
+
+		public override IDbConnection GetConnection()
+		{
+			IDbConnection con = null;
+
+			for (var i = 0; i < this.MaxTries; ++i)
```

Comment:
If `MaxTries <= 0`, this will yield `null` instead of a connection or an exception. Either we should test against `Math.Max(1, MaxTries)`, or setting `MaxTries` to zero or less should yield an exception.